packages/examples/src/chat-function-call.tsx

```diff
@@ -0,0 +1,65 @@
+import * as LLMx from 'ai-jsx';
+import { showInspector } from 'ai-jsx/core/inspector';
+import {
+  ChatCompletion,
+  ChatProvider,
+  SystemMessage,
+  UserMessage,
+  FunctionCall,
+  FunctionResponse,
+} from 'ai-jsx/core/completion';
+
+function ModelProducesFunctionCall({ query }: { query: string }) {
+  return (
+    <ChatProvider model="gpt-4-0613">
+      <ChatCompletion
+        functionDefinitions={[
+          {
+            name: 'evaluate_expression',
+            description: 'Evaluates a mathematical expression',
+            parameters: {
+              expression: {
+                description: 'The mathematical expression to be evaluated.',
+                type: 'string',
+                required: true,
+              },
+            },
+          },
+        ]}
+      >
+        <SystemMessage>You are a tool that may use functions to answer a user question.</SystemMessage>
+        <UserMessage>{query}</UserMessage>
+      </ChatCompletion>
+    </ChatProvider>
+  );
+}
+
+function ModelProducesFinalResponse({ query }: { query: string }) {
+  return (
+    <ChatProvider model="gpt-4-0613">
+      <ChatCompletion
+        functionDefinitions={[
+          {
+            name: 'evaluate_expression',
+            description: 'Evaluates a mathematical expression',
+            parameters: {
+              expression: {
+                description: 'The mathematical expression to be evaluated.',
+                type: 'string',
+                required: true,
+              },
+            },
+          },
+        ]}
+      >
+        <SystemMessage>You are a tool that may use functions to answer a user question.</SystemMessage>
+        <UserMessage>{query}</UserMessage>
+        <FunctionCall name="evaluate_expression" args={{ expression: '2523231 * 2382382' }} />
+        <FunctionResponse name="evaluate_expression">6011300116242</FunctionResponse>
+      </ChatCompletion>
+    </ChatProvider>
+  );
+}
+
+showInspector(<ModelProducesFunctionCall query="What is 2523231 * 2382382?" />);
+showInspector(<ModelProducesFinalResponse query="What is 2523231 * 2382382?" />);
```

Comment:
`showInspector` takes over the whole terminal, so we don't want to run two at once. I'd just leave one of these commented out, then the user can decide which to enable. Or you can make each one its own demo.

Reply:
If I comment one out I get a warning for the unused function that fails the test

Reply:
To work around that, you can either suppress that warning with an `// eslint-disable-next-line no-undef` comment, or you can make a separate file for each permutation.

Or, you could allow the user to pick on the command line, then use `process.argv` to decide which demo variant to run.

Reply:
Thanks! I did the `//eslint` trick since this file is just for demonstration not a real demo.